### PR TITLE
[baxtereus] support multiple coords for baxter ik

### PR DESCRIPTION
**Pull request by knorth55**
currently, baxter ik only supports sinlge target coordinates.
this PR supports multiple target coordinates.

### CODE REVIEW

File: jsk_baxter_robot/baxtereus/baxter-util.l [hunk 1]
```diff
@@ -37,21 +37,34 @@
      (setq r (send-super* :inverse-kinematics target-coords :avoid-collision-distance avoid-collision-distance :warnp warnp :dump-command dump-command args))
      (unless r ;;
        (format *error-output* "; failed for normal ik, starting from relaxed position~%")
-       (let  ((current-coords (send (cadr (memq :move-target args)) :copy-worldcoords)))
-         (send-super* :inverse-kinematics current-coords :rotation-axis nil :avoid-nspace-gain 0.1 :avoid-weight-gain 0.1 :stop 200 :avoid-collision-distance avoid-collision-distance args)
+       (let*  ((move-target (cadr (memq :move-target args)))
+               (current-coords (if (listp move-target)
+                                 (mapcar #'(lambda (x) (send x :copy-worldcoords)) move-target)
+                                 (send move-target :copy-worldcoords)))
+               (rotation-axis (if (listp current-coords) (make-list (length current-coords)) nil)))
```
Comment (pazeshun) [inline jsk_baxter_robot/baxtereus/baxter-util.l:44]:
```suggestion
       (let* ((move-target (cadr (memq :move-target args)))
              (current-coords (if (listp move-target)
                                (mapcar #'(lambda (x) (send x :copy-worldcoords)) move-target)
                                (send move-target :copy-worldcoords)))
              (rotation-axis (if (listp current-coords) (make-list (length current-coords)) nil)))
```

File: jsk_baxter_robot/baxtereus/baxter-util.l [hunk 2]
```diff
@@ -37,21 +37,34 @@
...
          ))
      (unless r ;;
        (format *error-output* "; failed for normal ik, try to move arms very slowly~%")
-       (let  ((step 0.0)
-              (current-coords (send (cadr (memq :move-target args)) :copy-worldcoords)))
+       (let*  ((step 0.0)
+               (move-target (cadr (memq :move-target args)))
+               (current-coords (if (listp move-target)
+                                 (mapcar #'(lambda (x) (send x :copy-worldcoords)) move-target)
+                                 (send move-target :copy-worldcoords))))
```
Comment (pazeshun) [inline jsk_baxter_robot/baxtereus/baxter-util.l:58]:
```suggestion
       (let* ((step 0.0)
              (move-target (cadr (memq :move-target args)))
              (current-coords (if (listp move-target)
                                (mapcar #'(lambda (x) (send x :copy-worldcoords)) move-target)
                                (send move-target :copy-worldcoords))))
```

**Review by pazeshun — CHANGES_REQUESTED** [review]

**Comment by knorth55** [comment]
@pazeshun sorry, but I found another bug in `:link-list`. `:link-list` is also considered as single link list. but we can use multiple link lists for euslisp ik interface.
I fixed the issued in https://github.com/jsk-ros-pkg/jsk_robot/pull/1349/commits/ad5637b39dc250a6d08ba8b749d08ad3f76c8011

**Review by pazeshun — CHANGES_REQUESTED** [review]
Sorry for late.
Could you add new tests to https://github.com/jsk-ros-pkg/jsk_robot/blob/25ac99ab82539b0e95d20470db1675b5f2f58c56/jsk_baxter_robot/baxtereus/test/test-baxter.l?
Because I failed to find the issue about `:link-list` and lost confidence...

**Comment by knorth55** [comment]
@pazeshun thanks, I update this PR for adding the test.

**Comment by knorth55** [comment]
Now, I split the test code in #1351.
#1351 will fail the test, and this PR will pass the test.

File: jsk_baxter_robot/baxtereus/test/test-baxter.l [hunk 3]
```diff
@@ -97,109 +188,65 @@
...
+        (do ((tmp-z (+ (- z (/ dz 2)) (/ step-dist 2)) (+ tmp-z step-dist)))
+          ((> tmp-z (+ z (/ dz 2))) t)
+          (setq pos (float-vector tmp-x tmp-y tmp-z))
+          (assert
+            (or
+              (send robot :rarm :inverse-kinematics
+                    (list (make-cascoords :pos pos))
+                    :move-target (list (send robot :larm :end-coords))
+                    :link-list
+                    (send robot :link-list (send (send robot :larm :end-coords) :parent))
```
Comment (pazeshun) [inline jsk_baxter_robot/baxtereus/test/test-baxter.l:225]: `(list (send robot :link-list (send (send robot :larm :end-coords) :parent)))`?

File: jsk_baxter_robot/baxtereus/test/test-baxter.l [hunk 4]
```diff
@@ -97,109 +188,65 @@
...
+                    :rotation-axis (list t)
+                    :debug-view (if *debug-view-p* :no-message nil))
+              (send robot :inverse-kinematics
+                    (list (make-cascoords :pos pos))
+                    :move-target (list (send robot :rarm :end-coords))
+                    :link-list
+                    (send robot :link-list (send (send robot :rarm :end-coords) :parent))
```
Comment (pazeshun) [inline jsk_baxter_robot/baxtereus/test/test-baxter.l:232]: `(list (send robot :link-list (send (send robot :rarm :end-coords) :parent)))`?

**Review by pazeshun — CHANGES_REQUESTED** [review]

**Comment by pazeshun** [comment]
At the time of #1177 , allowed test time of baxtereus is very limited, so adding many tests was unrealistic.
However, I think it is currently allowed because travis changed.
@knorth55 Is my understanding correct?

**Comment by knorth55** [comment]
@pazeshun thank you. I fixed the typo.
probably your understanding is correct.

**Review by pazeshun — APPROVED** [review]
Noetic test on github actions failed, but it has failed since #1354.

**Pull request merged**